aggregator/src/aggregator.rs

```diff
@@ -2427,16 +2438,18 @@ impl VdafOps {
     {
         datastore
             .run_tx(move |tx| {
+                let (task, collect_job_id) = (Arc::clone(&task), Arc::clone(&collect_job_id));
                 Box::pin(async move {
                     let collect_job = tx
                         .get_collect_job::<L, Q, A>(&collect_job_id)
                         .await?
                         .ok_or_else(|| {
                             datastore::Error::User(
-                                Error::UnrecognizedCollectJob(collect_job_id).into(),
+                                Error::UnrecognizedCollectJob(*collect_job_id).into(),
                             )
                         })?;
-
+                    Q::acknowledge_collection(tx, task.id(), collect_job.batch_identifier())
+                        .await?;
```

Comment:
FWIW this will make data inaccessible if the collector issues a DELETE request to a `current_batch` job it hasn't received yet, but I think that's a straightforward consequence of the DELETE semantics. (i.e. this is OK, and a collector should avoid doing that)

Reply:
That's a good point. I agree this is on-spec-but-funky behavior -- we could simply fail `DELETE` requests for a current batch, I can't think of a good reason to allow a collector to say "please nuke an arbitrary batch". I filed #846.